src/os/bluestore/BlueStore.cc

```diff
@@ -4012,6 +4012,12 @@ int BlueStore::get_block_device_fsid(CephContext* cct, const string& path,
 
 int BlueStore::_open_path()
 {
+  // sanity check(s)
+  if (cct->_conf->get_val<uint64_t>("osd_max_object_size") >=
```

Comment:
why not just use the `set_min_max()` and let the option validator do its job?

Reply:
the situation came up because a user with an actual cluster has objects > 4gb on filestore.  adding a hard limit will break their system.

Perhaps it is already broken, though, and we should force them to fix it before upgrading?  But in practice we don't have a way to prevent them from upgrading and breaking their cluster.